snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/core/ql/SnomedQlEvaluationRequest.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2018 B2i Healthcare Pte Ltd, http://b2i.sg
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.b2international.snowowl.snomed.core.ql;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nullable;
+
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.xtext.util.PolymorphicDispatcher;
+
+import com.b2international.index.query.Expression;
+import com.b2international.index.query.Expressions;
+import com.b2international.snowowl.core.domain.BranchContext;
+import com.b2international.snowowl.core.events.Request;
+import com.b2international.snowowl.core.events.util.Promise;
+import com.b2international.snowowl.eventbus.IEventBus;
+import com.b2international.snowowl.snomed.core.domain.SnomedDescription;
+import com.b2international.snowowl.snomed.core.ecl.EclExpression;
+import com.b2international.snowowl.snomed.core.ecl.EclSerializer;
+import com.b2international.snowowl.snomed.datastore.index.entry.SnomedConceptDocument;
+import com.b2international.snowowl.snomed.datastore.request.SnomedRequests;
+import com.b2international.snowowl.snomed.ecl.ecl.Script;
+import com.b2international.snowowl.snomed.ql.Conjunction;
+import com.b2international.snowowl.snomed.ql.Disjunction;
+import com.b2international.snowowl.snomed.ql.EclFilter;
+import com.b2international.snowowl.snomed.ql.Exclusion;
+import com.b2international.snowowl.snomed.ql.TermFilter;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Function;
+
+/**
+ * @since 6.12
+ */
+final class SnomedQlEvaluationRequest implements Request<BranchContext, Promise<Expression>> {
+
+	private static final long serialVersionUID = 8932162693072727864L;
+
+	private final PolymorphicDispatcher<Promise<Expression>> dispatcher = PolymorphicDispatcher.createForSingleTarget("eval", 2, 2, this);
```

Comment:
Make this variable transient and add `JsonIgnore` annotation as well, so it won't get serialized through the network.